src/main/java/com/hierynomus/smbj/share/DiskShare.java

```diff
@@ -466,6 +468,127 @@ private boolean exists(String path, EnumSet<SMB2CreateOptions> createOptions) th
         }
     }
 
+    /***
+     * Rename(Move) Directory/File. Default will using the false for replaceIfExist option
+     * @param path the path to the original directory/file
+     * @param newFileName the new directory/file name/path after rename
+     * @throws TransportException
+     * @throws SMBApiException
+     */
+    public void rename(String path, String newFileName) throws TransportException, SMBApiException {
```

Comment:
Ok lol, I got what you mean now. Actually I didn't think too deep on why I putting it on the DiskShare but these are the reasons:

1. Able to reuse the setInfoCommon 
I want to implement for the set methods for `FileBasicInformation` at the beginning. I found the only usage for `SMB2QueryInfoRequest` and the `queryInfoCommon` in the DiskShare. I simply think the getter and setter should in the same class and I think I better create a method for setInfoCommon, just like the `queryInfoCommon` and can reuse for those set methods. After that, I just simply think the rename operation can also reuse the function so I put it in `DiskShare`.

2.  Convenient to calling it in DiskShare
It doesn't seems convenience to put the it in the `File` and `Directory` classes. In my opinion, usually we won't perform other operations when renaming the file. Also, the rename operation required `DELETE` access right to perform the rename. I think most of the programs will not grant the `DELETE` access right by default. That means if we put it in `File` and `Directory` class, we usually got somethings like this : 
```
file.close()
file = share.openFile(... , toLong(EnumSet.of(GENERIC_WRITE, DELETE)), ...)
file.rename(newpath)
file.close()
```
Of course, we can simply create a `mv` method for the `DiskShare` and call the `file.rename` to solve this problem, but I just think most of the use case will just simply calling the `DiskShare.mv()` / `DiskShare.rename()` method so I just simply put it in `DiskShare`.